go/vt/sqlparser/sql.y

```diff
@@ -2331,6 +2370,7 @@ reserved_keyword:
 | LOCALTIMESTAMP
 | LOCK
 | MATCH
+| MAXVALUE
```

Comment:
i was following the guidance of:
https://dev.mysql.com/doc/refman/5.7/en/keywords.html

moving MAXVALUE causes 1 shift/reduce, but moving PARTITION to non_reserved produces no conflicts, compiles, and passes tests.